File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []

    for x in list(s):
        if x == '(':
            stack.append('(')
        else:
            if len(stack) == 0:
                return False
            stack.pop()
    return True if len(stack) == 0 else False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Deque<Character> stack = new ArrayDeque<Character>();
        if(s.length() % 2 == 1 || s.charAt(0) == ')') return false;
        for(int i = 0 ; i < s.length() ; i++){
            if(stack.isEmpty()){
                stack.offer(s.charAt(i));
                continue;
            }
            if(stack.peek() == '(' && s.charAt(i) == ')'){
                stack.pop();
            }else {
                stack.offer(s.charAt(i));
            }
        }
        return stack.isEmpty();
    }
}